nltk/collocations.py

```diff
@@ -54,13 +54,14 @@ class AbstractCollocationFinder(object):
     def __init__(self, word_fd, ngram_fd):
         self.word_fd = word_fd
         self.ngram_fd = ngram_fd
-
+    
     @classmethod
     def from_documents(cls, documents):
         """Constructs a collocation finder given a collection of documents,
         each of which is a list (or iterable) of tokens.
         """
-        return cls.from_words(_itertools.chain(*documents))
+        #return cls.from_words(_itertools.chain(*documents))
+        return cls.from_new_documents(documents) 
```

Comment:
Rather than this awkward indirection, you could store the default window size as a class attrib.